libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/VerifiableCredential.java

```diff
@@ -15,28 +20,48 @@
 @Data
 @EqualsAndHashCode(exclude = "signedJwt")
 public class VerifiableCredential {
+    private static final Logger LOGGER = LogManager.getLogger();
+
     private final String userId;
     private final String criId;
     private final String vcString;
     private final JWTClaimsSet claimsSet;
     private final SignedJWT signedJwt;
     private Instant migrated;
+    private final uk.gov.di.model.VerifiableCredential credential;
 
     private VerifiableCredential(String userId, String criId, SignedJWT signedJwt, Instant migrated)
             throws CredentialParseException {
         try {
+            var claimsSet = signedJwt.getJWTClaimsSet();
             this.userId = userId;
             this.criId = criId;
             this.vcString = signedJwt.serialize();
-            this.claimsSet = signedJwt.getJWTClaimsSet();
+            this.claimsSet = claimsSet;
             this.signedJwt = signedJwt;
             this.migrated = migrated;
+            this.credential = parseCredential(claimsSet);
         } catch (ParseException e) {
             throw new CredentialParseException(
                     "Failed to get jwt claims to construct verifiable credential", e);
         }
     }
 
+    private uk.gov.di.model.VerifiableCredential parseCredential(JWTClaimsSet claimsSet)
+            throws CredentialParseException {
+        try {
+            return VerifiableCredentialParser.parseCredential(claimsSet);
+        } catch (CredentialParseException e) {
+            if ("production".equals(System.getenv(EnvironmentVariable.ENVIRONMENT.name()))) {
```

Comment:
This isn't ideal, but it's quite hard to get a config service here and I'm trying to be particularly paranoid about it!

I expect to remove this in the first PR that actually uses the parsed credential.